crates/rattler_installs_packages/src/artifacts/sdist.rs

```diff
@@ -65,24 +66,33 @@ impl SDist {
     /// Find entry in tar archive
     fn find_entry(&self, name: impl AsRef<Path>) -> std::io::Result<Option<Vec<u8>>> {
         let mut lock = self.file.lock();
-        let mut archive = generic_archive_reader(&mut lock, self.name.format)?;
-
-        fn skip_first_component(path: &Path) -> PathBuf {
-            path.components().skip(1).collect()
-        }
-
-        // Loop over entries
-        for entry in archive.entries()? {
-            let mut entry = entry?;
-
-            // Find name in archive and return this
-            if skip_first_component(entry.path()?.as_ref()) == name.as_ref() {
+        let archives = generic_archive_reader(&mut lock, self.name.format)?;
+        match archives {
+            Archives::TarArchive(mut archive) => {
+                fn skip_first_component(path: &Path) -> PathBuf {
+                    path.components().skip(1).collect()
+                }
+
+                // Loop over entries
+                for entry in archive.entries()? {
+                    let mut entry = entry?;
+
+                    // Find name in archive and return this
+                    if skip_first_component(entry.path()?.as_ref()) == name.as_ref() {
+                        let mut bytes = Vec::new();
+                        entry.read_to_end(&mut bytes)?;
+                        return Ok(Some(bytes));
+                    }
+                }
+                Ok(None)
+            }
+            Archives::Zip(mut archive) => {
+                let mut zip_file = archive.by_name(name.as_ref().to_str().unwrap())?;
```

Comment:
Do zip archives not have an "inner" folder? In the `tar` archives we do the `skip_first_component` thing.

Reply:
With tar.gz files, the folder structure is something like: `mypkg-0.1.0/pyproject.toml`
which is why we do the `strip_component` stuff.

So I guess I was wondering if that is also the case for zip sdists.

Reply:
Hmm, they do look very similar directory-layout wise with the parent-folder being the <name>-<version>. Unsure why this works! :)

╭─ pixi ~/Downloads
╰─$ unzip -l django-viewclass-mixins-1.1.3.zip
Archive:  django-viewclass-mixins-1.1.3.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1565  07-10-2012 17:05   django-viewclass-mixins-1.1.3/LICENSE
      100  12-02-2012 16:21   django-viewclass-mixins-1.1.3/MANIFEST.in
     2216  12-09-2014 12:41   django-viewclass-mixins-1.1.3/PKG-INFO
     1437  09-18-2014 14:09   django-viewclass-mixins-1.1.3/README.md
       12  05-17-2013 11:57   django-viewclass-mixins-1.1.3/requirements.txt
       64  12-09-2014 12:41   django-viewclass-mixins-1.1.3/setup.cfg
     1351  11-02-2012 16:06   django-viewclass-mixins-1.1.3/setup.py
      132  05-17-2013 11:36   django-viewclass-mixins-1.1.3/tox.ini
        1  12-09-2014 12:41   django-viewclass-mixins-1.1.3/django_viewclass_mixins.egg-info/dependency_links.txt
     2216  12-09-2014 12:41   django-viewclass-mixins-1.1.3/django_viewclass_mixins.egg-info/PKG-INFO
       12  12-09-2014 12:41   django-viewclass-mixins-1.1.3/django_viewclass_mixins.egg-info/requires.txt
      427  12-09-2014 12:41   django-viewclass-mixins-1.1.3/django_viewclass_mixins.egg-info/SOURCES.txt
       17  12-09-2014 12:41   django-viewclass-mixins-1.1.3/django_viewclass_mixins.egg-info/top_level.txt
      449  08-31-2012 17:25   django-viewclass-mixins-1.1.3/docs/DeactivateMixin.md
      694  08-31-2012 17:25   django-viewclass-mixins-1.1.3/docs/FilteredListMixin.md
      885  08-31-2012 17:33   django-viewclass-mixins-1.1.3/docs/ObjectOwnerMixin.md
     7990  12-09-2014 12:40   django-viewclass-mixins-1.1.3/viewclass_mixins/views.py
      601  12-09-2014 12:40   django-viewclass-mixins-1.1.3/viewclass_mixins/__init__.py
---------                     -------
    20169                     18 files
╭─ pixi ~/Downloads
╰─$ unzip -l configmy-0.14.87
Archive:  configmy-0.14.87.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      207  09-20-2017 00:31   configmy-0.14.87/MANIFEST.in
     1349  10-28-2017 12:21   configmy-0.14.87/PKG-INFO
      502  09-20-2017 17:18   configmy-0.14.87/README.md
       64  10-28-2017 12:21   configmy-0.14.87/setup.cfg
     4256  10-28-2017 12:21   configmy-0.14.87/setup.py
    67556  10-06-2017 17:05   configmy-0.14.87/configmy/codesource.py
    20162  10-06-2017 14:50   configmy-0.14.87/configmy/configmy.py
     1307  09-29-2017 22:44   configmy-0.14.87/configmy/__init__.py
        1  10-28-2017 12:21   configmy-0.14.87/configmy.egg-info/dependency_links.txt
     1349  10-28-2017 12:21   configmy-0.14.87/configmy.egg-info/PKG-INFO
      527  10-28-2017 12:21   configmy-0.14.87/configmy.egg-info/SOURCES.txt
        9  10-28-2017 12:21   configmy-0.14.87/configmy.egg-info/top_level.txt
      514  09-20-2017 00:31   configmy-0.14.87/pkg/copr-build.sh
      115  09-20-2017 00:31   configmy-0.14.87/pkg/entry_points.txt
      437  09-20-2017 00:31   configmy-0.14.87/pkg/gen-authors.sh
      283  09-20-2017 00:31   configmy-0.14.87/pkg/gen-readme.sh
      205  09-20-2017 00:31   configmy-0.14.87/pkg/nose.cfg
    21007  09-20-2017 00:31   configmy-0.14.87/pkg/package.spec.in
      180  09-20-2017 00:31   configmy-0.14.87/pkg/requirements_full-py-2.6.txt
      111  09-20-2017 00:31   configmy-0.14.87/pkg/requirements_full.txt
     2049  09-20-2017 00:31   configmy-0.14.87/pkg/rpmbuild-wrapper.sh
     1891  09-20-2017 00:31   configmy-0.14.87/pkg/runtest.sh
      157  09-20-2017 00:31   configmy-0.14.87/pkg/scrutinizer.yml
      264  09-20-2017 00:31   configmy-0.14.87/pkg/test_requirements-py-2.6.txt
      137  09-20-2017 00:31   configmy-0.14.87/pkg/test_requirements.txt
        0  09-20-2017 00:31   configmy-0.14.87/tests/__init__.py
---------                     -------
   124639                     26 files
╭─ pixi ~/Downloads

Reply:
thank you for the observation
let me test it more closer

Reply:
Other than this looks fine to me!

Reply:
indeed it does not find the file using direct archive.by_name approach. I've change the implementation and added 2 tests to validate that it find file for both zip and .tar